src/adapters/supabase/helpers/tables/access.ts

```diff
@@ -0,0 +1,75 @@
+import { SupabaseClient } from "@supabase/supabase-js";
+import { Database } from "../../types/database";
+import { GitHubNode } from "../../types/github";
+import { Super } from "./super";
+import { UserRow } from "./user";
+import { Context } from "../../../../types/context";
+import { Comment } from "../../../../types/github";
+
+type AccessRow = Database["public"]["Tables"]["access"]["Row"];
+type AccessInsert = Database["public"]["Tables"]["access"]["Insert"];
+type UserWithAccess = UserRow & { access: AccessRow[] };
+
+type AccessData = {
+  user_id: number;
+  multiplier: number;
+  multiplier_reason: string;
+  node_id: string;
+  node_type: string;
+  node_url: string;
+};
+
+export class Access extends Super {
+  constructor(supabase: SupabaseClient, context: Context) {
+    super(supabase, context);
+  }
+
+  private async _getUserWithAccess(id: number): Promise<UserWithAccess> {
+    const { data, error } = await this.supabase.from("users").select("*, access(*)").filter("id", "eq", id).single();
+
+    if (error) {
+      this.context.logger.fatal(error.message, error);
+      throw new Error(error.message);
+    }
+    return data;
+  }
+
+  public async getAccess(id: number): Promise<AccessRow | null> {
+    const userWithAccess = await this._getUserWithAccess(id);
+    if (userWithAccess.access.length === 0) {
+      this.context.logger.debug("No access found for user", { id });
+      return null;
+    }
+    return userWithAccess.access[0];
+  }
+
+  public async setAccess(labels: string[], node: GitHubNode, userId?: number): Promise<null> {
+    const { data, error } = await this.supabase.from("access").upsert({
+      labels: labels,
+      ...node,
+      user_id: userId,
+    } as AccessInsert);
+    if (error) throw new Error(error.message);
+    return data;
+  }
+
+  async upsertMultiplier(userId: number, multiplier: number, reason: string, comment: Comment) {
```

Comment:
I'm wondering if multiplier should be a part of this plugin because conversation-rewards plugin will need to access the multiplier

Reply:
We can remove this. I think that it should be in the permit generation module as a low priority capability because we don't use it anymore and to be honest it was barely used.